src/main/java/org/spongepowered/common/data/persistence/ConfigurateTranslator.java

```diff
@@ -75,64 +73,56 @@ private static DataContainer translateFromNode(ConfigurationNode node) {
+            } else if (n.hasListChildren()) {
+                translate(n, container);
+            } else {
+                DataContainer sub = DataContainer.createNew(DataView.SafetyMode.NO_DATA_CLONED);
+                translate(n, sub);
+                container.set(of(String.valueOf(o.getKey())), sub);
+            }
         }
     }
 
-    private List<Object> translate(List<? extends ConfigurationNode> node, DataView container, List<?> value) {
+    private void translate(List<? extends ConfigurationNode> node, DataView container, Object key, List<?> value) {
         List<Object> list = new ArrayList<>(value.size());
         for (int i = 0; i < value.size(); i++) {
             ConfigurationNode n = node.get(i);
-            if (n.getValueType() == ValueType.SCALAR) {
+            if (!n.getValueType().canHaveChildren()) {
                 list.add(n.getValue());
-            } else if (n.hasMapChildren()) {
-                DataContainer clean = DataContainer.createNew(DataView.SafetyMode.NO_DATA_CLONED);
-                translate(n.getChildrenMap(), clean, (Map<?, ?>) n.getValue());
-                list.add(clean);
             } else {
-                DataContainer clean = DataContainer.createNew(DataView.SafetyMode.NO_DATA_CLONED);
-                translate(n, clean);
-                list.add(clean);
+                DataContainer sub = DataContainer.createNew(DataView.SafetyMode.NO_DATA_CLONED);
+                translate(n, sub);
+                list.add(sub);
             }
         }
-        return list;
+        container.set(of(String.valueOf(key)), list);
     }
 
     private void translate(ConfigurationNode node, DataView container) {
         Object key = node.getKey();
         Object value = node.getValue();
         if (node.hasMapChildren()) {
-            DataView view;
-            if (key != null) {
-                view = container.createView(of(key.toString()));
-            } else {
-                view = container;
-            }
-            translate(node.getChildrenMap(), view, (Map<?, ?>) value);
-        } else if (node.hasListChildren()) {
-            List<Object> list = translate(node.getChildrenList(), container, (List<?>) node.getValue());
-            container.set(of(String.valueOf(key)), list);
-        } else {
-            container.set(of('.', String.valueOf(node.getKey())), node.getValue());
+            translate(node.getChildrenMap(), container, key, (Map<?, ?>) value);
+            return;
         }
-    }
 
-    public ConfigurationNode translateData(DataView container) {
-        ConfigurationNode node = SimpleConfigurationNode.root();
-        translateContainerToData(node, container);
-        return node;
+        if (node.hasListChildren()) {
+            translate(node.getChildrenList(), container, key, (List<?>) value);
+            return;
+        }
+
+        container.set(of('.', String.valueOf(key)), value);
```

Comment:
Configurate uses `.` as a separator itself, it will never be in the key of a node.

Additionally, I believe a test case for a root node holding only a value (such as `true`) will fail here as the key is null, which gets converted to a path consisting of the string `null`. This is the actual problem in the issue.

Reply:
> Configurate uses . as a separator itself, it will never be in the key of a node.

That's wrong, it actually doesn't have a separator, which is why the path to a node is represented as `Object[]`. A `.` can actually be in the key of a node without being a separator. If you want an example of this, check Nucleus' `commands.conf` file.

> Additionally, I believe a test case for a root node holding only a value (such as true) will fail here as the key is null, which gets converted to a path consisting of the string null. This is the actual problem in the issue.

That is true, though I'm not sure the path that processes the root node will actually end up down here.

@ImMorpheus [The test case here](https://github.com/SpongePowered/SpongeCommon/pull/2353/files#diff-c5975971207a43e7feaecf528c5b4ba8R104) should be expanded to ensure that the end result is what we expect - that'll tell is if this is a problem.

Reply:
Hmm, maybe this is outdated but I've always thought that calls to `getNode("this.that.other")` would split on the `.` automatically. Either way, I did forget about the case of quoted keys which could be use. However, I would say that taking the splitting on this is still not the right decision because the end result should match the structure of the node.

I was not able to find a javadoc specifying the behavior of child nodes for `DataView`, but I can try to explain the issues that would happen if the key were used. Because a `DataView` also has a path, we can link the structure of it to that to `ConfigurationNode`s and see what happens when serializing and deserializing.

When a value is set, that value is expected to be deserializeable along the given key. If I set a color to the node `my-color`, I expect something like `my-color { red=255, green=170, blue=0 }`. If I try to deserialize a color from this entire thing, it would fail - you need to get the value from the `my-color` node. If the key is included in the path during deserialization, I would get back a `DataContainer` containing `my-color { red=255, green=170, and blue=0}`. This `DataContainer` represents the color *value* (note that it's picked up an additional key that is user-specific), and is thus when set in a larger `DataView` under `my-color` the key becomes duplicated, resulting in `my-color { my-color { red=255, green=170, blue=0 } }`. Apologies if this was confusing but I tried xD

As additional example of this, [here is similar logic from Configurate](https://github.com/SpongePowered/Configurate/blob/master/configurate-core/src/main/java/ninja/leaping/configurate/SimpleConfigurationNode.java#L244) when setting a value to a `ConfigurationNode` - the key of the node is ignored and the value is set directly.

Reply:
Okay, what I meant was what was the previous behaviour. That is, what is expected by plugins now? At this stage in the 7 cycle, it should be that, plus whatever is needed to fix the problems that were there.

Reply:
Ah, gotcha. My belief is the that the current behavior is inconsistent; it will deserialize to a different format than serialized from when using non-root nodes. The `populateNode` method, for example, is only ever called with a root. I will be able to write some tests later today to confirm this.